tests/test_async.py

```diff
@@ -24,6 +24,20 @@ async def test_wait_during_not_done(sleeper):
         sleeper.sleep_for_a_bit())
 
 
+@pytest.mark.asyncio
+async def test_wait_until_done_with_delay(sleeper):
```

Comment:
you can just merge this test with 'test_wait_until_done' it's basically the same